File: model/src/main/java/io/sparkled/model/validator/SequenceValidator.java
package io.sparkled.model.validator;

import io.sparkled.model.entity.Sequence;
import io.sparkled.model.validator.exception.EntityValidationException;

public class SequenceValidator {

    public void validate(Sequence sequence) {
        if (sequence.getName() == null) {
            throw new EntityValidationException(Errors.NAME_MISSING);
        } else if (sequence.getFramesPerSecond() <= 0) {
            throw new EntityValidationException(Errors.FRAMES_PER_SECOND_NOT_POSITIVE);
        }
    }

    private static class Errors {
        static final String NAME_MISSING = "Sequence name must not be empty.";
        static final String FRAMES_PER_SECOND_NOT_POSITIVE = "Sequence frames per second must be greater than 0.";
    }
}

File: model/src/main/kotlin/io/sparkled/model/validator/SequenceValidator.kt
package io.sparkled.model.validator

import io.sparkled.model.SequenceModel
import io.sparkled.model.validator.exception.EntityValidationException

class SequenceValidator {

    fun validate(sequence: SequenceModel) {
        if (sequence.framesPerSecond <= 0) {
            throw EntityValidationException(Errors.FRAMES_PER_SECOND_NOT_POSITIVE)
        }
    }

    private object Errors {
        const val FRAMES_PER_SECOND_NOT_POSITIVE = "Sequence frames per second must be greater than 0."
    }
}
